plugins/out_s3/s3.c

```diff
@@ -1524,6 +1678,16 @@ static int cb_s3_exit(void *data, struct flb_config *config)
         }
     }
 
+    if (remove(ctx->seq_index_file) != 0) {
+        flb_plg_error(ctx->ins, "Could not successfully delete metadata file on exit. "
+                      "Overwriting index to 0");
+        ret = write_seq_index(ctx->seq_index_file, 0, 0);
+        if (ret < 0) {
+            flb_plg_error(ctx->ins, "Failed to overwrite index to 0");
+            return -1;
+        }
+    }
```

Comment:
is overwriting the index file on exit the desired behavior? I thought the whole point of saving the index in a file is so that it could survive restarts. If you delete/overwrite it on shut down then there's no point in having the file I think..